pkg/registry/etcd/etcd.go

```diff
@@ -347,7 +347,17 @@ func (r *Registry) WatchControllers(ctx api.Context, label, field labels.Selecto
 		}
 		match := label.Matches(labels.Set(controller.Labels))
 		if match {
-			pods, _ := r.ListPods(ctx, labels.Set(controller.Spec.Selector).AsSelector())
+			pods, err := r.ListPods(ctx, labels.Set(controller.Spec.Selector).AsSelector())
+			if err != nil {
+				glog.Warningf("Error listing pods: %v", err)
+				// No object that's useable so drop it on the floor
+				return false
+			}
+			if pods != nil {
+				glog.Warningf("Pods list is nil.  This should never happen...")
```

Comment:
oops, good catch, should be ```pods == nil``` not ```!=``` copy/paste error from the above...

pods.Items might be nil, but pods certainly should _not_ be nil unless err is non-nil, which I'm checking above.

See the stack trace in #3624 which is a nil dereference of a nil ```pods``` variable.  In that case, I'm pretty sure ```err``` was non nil, but as long as I was in here, I figured I'd add the extra check, just in case.